### PR TITLE
Issue 45498: Show cross-linked peptide sequence info

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
To better support multi attribute method analysis, Skyline added support for cross-linked peptides a couple of years ago. These are handled as special peptide modification annotations. Panorama can show the string representation but would greatly benefit from a more intuitive rendering, where we show all of the peptides involved in the linking and highlight the amino acids involved in the linking. We also want to improve the peptide mapping report for MAM folders to show more details about how and where the peptides are linked.

#### Changes
* Parse cross linked peptide sequences into a reusable data structure
* Render all peptides with link locations as modifications
* Unit tests for parsing, Selenium tests for peptide map rendering
* New/updated columns in Peptide Map report for proteins involved in linking and location of links
* More flexible column filtering for custom, MAM-related PTM report

### CODE REVIEW

File: src/org/labkey/targetedms/query/SamplePivotCustomizer.java [hunk 1]
```diff
@@ -33,92 +31,157 @@
...
-            throw new IllegalArgumentException("Must have exactly one property named 'runIdField'");
+            throw new IllegalArgumentException("Must have exactly one property named '" + propertyName + "'");
         }
-        _runIdField = FieldKey.fromString(runIdFilter.iterator().next());
+        return null;
     }
 
-    @Override
-    public void customize(TableInfo tableInfo)
+    private Integer getRunId()
```
Comment (ankurjuneja) [inline src/org/labkey/targetedms/query/SamplePivotCustomizer.java]: As runIds are of type long now, better to be consistent across.

File: src/org/labkey/targetedms/query/SamplePivotCustomizer.java [hunk 2]
```diff
@@ -33,92 +31,157 @@
...
+                    clause.getParamVals().length == 1 &&
+                    clause.getParamVals()[0] != null)
             {
-                // Grab any filters that have been applied. Use QuerySettings, which will find the right values whether it's
-                // a GET or POST
-                QuerySettings settings = new QuerySettings(context, "query");
-                filterURL = settings.getSortFilterURL();
+                try
+                {
+                    // Get the sample names associated with that run
```
Comment (ankurjuneja) [inline src/org/labkey/targetedms/query/SamplePivotCustomizer.java]: I am confused with the comment here.

**Review by ankurjuneja — APPROVED** [review]

File: src/org/labkey/targetedms/query/SamplePivotCustomizer.java (hunk 1 above)
Comment (labkey-jeckels) [reply src/org/labkey/targetedms/query/SamplePivotCustomizer.java]: We're a long way from 2 billion runs uploaded but agree best to parse as the right data type. Updated

File: src/org/labkey/targetedms/query/SamplePivotCustomizer.java (hunk 2 above)
Comment (labkey-jeckels) [reply src/org/labkey/targetedms/query/SamplePivotCustomizer.java]: Fixed. Agreed that it didn't make sense where it landed after the refactor

**Pull request merged**